src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,77 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import debounce from 'lodash.debounce';
+import React, { useMemo, useState } from 'react';
+
+interface Props {
+  onSelect?: (status: string) => void;
+  onChange?: (query: string) => void;
+}
+
+export const TodoFilter: React.FC<Props> = ({ onSelect, onChange }) => {
+  const [value, setValue] = useState('');
+
+  const applyFilter = useMemo(
+    () =>
+      debounce((currentQuery: string) => {
+        if (onChange) {
+          onChange(currentQuery);
+        }
+      }, 300),
+    [onChange],
+  );
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    applyFilter(event.target.value);
+    setValue(event.target.value);
+  };
+
+  const handleQueryDelete = () => {
+    applyFilter('');
+    setValue('');
```

Comment:
The `handleQueryDelete` function is correctly clearing the input and applying the filter. However, you can simplify the `onClick` handler in the button by directly passing `handleQueryDelete` without wrapping it in an arrow function.